core/src/main/java/com/yahoo/imapnio/async/data/ExtendedModifiedSinceTerm.java

```diff
@@ -0,0 +1,127 @@
+ *                       ;; used by "quoted" string as described in
+ *                       ;; Section 4.3 of [RFC3501]; e.g., for the
+ *                       ;; flag \Seen, the corresponding <entry-name>
+ *                       ;; is "/flags/\\seen", and for the flag
+ *                       ;; $MDNSent, the corresponding <entry-name>
+ *                       ;; is "/flags/$mdnsent".
+ *
+ * entry-type-resp     = "priv" / "shared"
+ *                       ;; Metadata item type.
+ *
+ * entry-type-req      = entry-type-resp / "all"
+ *                       ;; Perform SEARCH operation on a private
+ *                       ;; metadata item, shared metadata item,
+ *                       ;; or both.
+ *
+ * }
+ * </pre>
+ */
+public final class ExtendedModifiedSinceTerm extends SearchTerm {
+
+    /** The name of the metadata item. */
+    private final Flags entryName;
+
+    /** The type of metadata item. */
+    private final EntryTypeReq entryType;
+
+    /** The given modification sequence. */
+    private final long modSeq;
+
+    /**
+     * Constructor.
+     *
+     * @param modSeq modification sequence number
+     */
+    public ExtendedModifiedSinceTerm(final long modSeq) {
+        this.entryName = null;
+        this.entryType = null;
+        this.modSeq = modSeq;
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param entryName name of the metadata item
+     * @param entryType type of the metadata item
+     * @param modSeq modification sequence number
+     * @throws ImapAsyncClientException if entryName has more than one flag.
+     */
+    public ExtendedModifiedSinceTerm(@Nonnull final Flags entryName, @Nonnull final EntryTypeReq entryType, final long modSeq)
+            throws ImapAsyncClientException {
+        final Flags.Flag[] sf = entryName.getSystemFlags(); // get the system flags
+        final String[] uf = entryName.getUserFlags(); // get the user flag strings
+        if (sf.length + uf.length != 1) {
+            throw new ImapAsyncClientException(ImapAsyncClientException.FailureType.INVALID_INPUT);
+        }
+        this.entryName = entryName;
+        this.entryType = entryType;
+        this.modSeq = modSeq;
+    }
+
+    /**
+     * @return the entry name
+     */
+    @Nullable
+    public Flags getEntryName() {
+        return entryName;
+    }
+
+    /**
+     * @return the entry type
+     */
+    @Nullable
+    public EntryTypeReq getEntryType() {
+        return entryType;
+    }
+
+    /**
+     * @return the modification sequence
+     */
+    public long getModSeq() {
+        return modSeq;
+    }
+
+    /**
+     * This method should never be called for ExtendedModifiedSinceTerm.
+     *
+     * @param msg the date comparator is applied to this Message's MODSEQ
+     * @return UnsupportedOperationException if the method is called
+     */
+    @Override
+    public boolean match(final Message msg) {
+        throw new UnsupportedOperationException();
```

Comment:
Is it possible put throws statement and javadoc especially it is a runtime exception?

Reply:
No. Check style and CI don't allow me to write throw in javadoc.

Reply:
Can you put some error message inside UnsupportedOperationException()?